### DISCUSSION TITLE
How to set passive to false on an event listener

### DISCUSSION THREAD

**Discussion by steckes** [leptos-rs/leptos#1791] [Q&A]
Hey!
For plot interaction I need to call event.prevent_default() on the wheel event.
I just created a callback and connected it to the component with on:wheel.
Now the browser does not care about the prevent default, because the event is declared as passive.
Is there a way to declare the event as passive = false? The only thing I found is the following in web_sys:

`HtmlElement::add_event_listener_with_callback_and_add_event_listener_options("wheel", <js_sys::Function>, AddEventListenerOptions::passive(false))`

But there I don't know how to add my callback as js_sys::Function. Is there a different way of achieving this?

Thank you :)
I'm planning on open sourcing the plotter as a leptos component as soon as the basics are done!

**Comment by gbj** [comment]
There's nothing built into the framework that touches on this.

As you note you can do it with `web-sys` directly. For information on using callbacks see the `wasm-bindgen` docs:
https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/closure/struct.Closure.html#casting-a-closure-to-a-js_sysfunction
https://rustwasm.github.io/docs/wasm-bindgen/examples/closures.html

**Reply by steckes** [reply]
Cool thanks for the hint!
I'm anyways super hyped that through web_sys there is nothing you can't do :)
